src/main/java/de/retest/recheck/TestCaseFinder.java

```diff
@@ -86,7 +88,41 @@ public static TestCaseFinder getInstance() {
 	 * @return The class name for the test case method in all stack traces.
 	 */
 	public Optional<String> findTestCaseClassNameInStack() {
-		return findTestCaseMethodInStack( toClassName() );
+		final Optional<String> testCaseClassName = findTestCaseMethodInStack( toClassName() );
+		if ( testCaseClassName.isPresent() ) {
+			return testCaseClassName;
+		} else {
+			return findTestCaseClassInStack();
+		}
```

Comment:
Now we traverse the stack trace twice (in the worst case). This is an expensive operation, that's why they introduced the Stack-Walking API in 9.

Does it make sense to unify these methods?

Reply:
There is public `findTestCaseMethodInStack()`.

Reply:
I thought about this too, but this would require some refactoring. I'm really not sure if this is worth the effort …